### DISCUSSION TITLE
When using pdf.render_file, I want to save the file to amazon S3.

### DISCUSSION THREAD

**Discussion by akkyta** [prawnpdf/prawn#1221] [Q&A]
How should I write to specify the save destination?

pdf.render_file "test.pdf"
This code will be saved in the root directory.

I spend many days on this difficult problem.
It would be really helpful if you could tell me.

**Comment by pointlessone** [comment]
You can use directories in the file name argument. It can be either relative or full path.

```ruby
pdf.render_file "subdirectory/test.pdf"
```

Prawn does not provide anything to save to S3 or anything but local file system. You'll have to consult documentation of the library you use to communicate with S3 to upload the generated file to S3.

**Comment by fidothe** [comment]
If you're using AWS's official Ruby client, you can use the `render` method on a `Prawn::Document` to render to the PDF as a string you can pass to the `put_object` method of `Aws::S3::Client`:

```ruby

pdf = Prawn::Document.new
pdf.text "Hello World"

s3_client = Aws::S3::Client.new
s3_client.put_object({
  body: pdf.render, 
  bucket: "examplebucket", 
  key: "exampleobject"
})
```

See the AWS SDK docs at https://docs.aws.amazon.com/sdk-for-ruby/v3/api/index.html for more details.
